R/cchart.R

```diff
@@ -844,6 +843,20 @@ px2pt <- function(x)
     return(x/1.3333)
 }
 
+getHexCode <- function(color, opacity)
+{
+    if (!(opacity >= 0 && opacity <= 1))
+        return(color)
+    if (substr(color, 0, 1) == '#' && nchar(color) == 7)
+        return(sprintf("%s%02X", color, round(opacity * 255)))
+    
+    # Returns grey if color is not recognized - matches checkColors
+    tmp <- try(col2rgb(color, alpha = TRUE), silent)
+    if (inherits(tmp, "try-error"))
+        return ('#CCCCCC')
+    # If color is 8-digit hex than multiply opacity - matches plotly::toRGB
+    return(rgb(t(tmp), alpha = round(opacity * tmp["alpha",]), maxColorValue = 255))
```

Comment:
Could a more informative name than `tmp` be used?
maybe `col.as.rgb`?

`silent` isn't defined (should be `silent = TRUE`?)

Maybe something like?
```suggestion
    col.as.rgb <- try(t(col2rgb(color, alpha = TRUE)), silent = TRUE)
    if (inherits(col.as.rgb, "try-error"))
        return("#CCCCCC")
    # If color is 8-digit hex than multiply opacity - matches plotly::toRGB
    return(rgb(col.as.rgb, alpha = round(opacity * col.as.rgb[, "alpha"]), maxColorValue = 255))
```